src/device-state.ts

```diff
@@ -23,6 +23,7 @@ import {
 	UpdatesLockedError,
 } from './lib/errors';
 import * as updateLock from './lib/update-lock';
+import { readLock, writeLock } from './lib/process-lock';
```

Comment:
Is kind of weird that the state engine uses the same lock instance as the target state or the update lock. I may be misunderstanding how `rwlock` operates

Reply:
This is how rwlock works, by locking arbitrarily defined resources AKA strings. Device state locks the `'target'` resource:
https://github.com/balena-os/balena-supervisor/blob/d6efb0ba03142b628ce0e321dcbabc2a7cb8f6fb/src/device-state.ts#L296-L301

I agree, it's definitely confusing .